test/gurka/test_route.cc

```diff
@@ -636,3 +636,46 @@ TEST(Standalone, DoNotAllowDoubleUturns) {
       FAIL() << "Unexpected path found!";
   } catch (const std::exception& e) { EXPECT_STREQ(e.what(), "No path could be found for input"); }
 }
+
+TEST(Standalone, OneWayIdToManyEdgeIds) {
+  const std::string ascii_map = R"(
+        A<--B<--C<---D
+                ^
+                |
+                E
+                ^
+                |
+                F
+)";
+
+  const gurka::ways ways = {
+      {"FE", {{"highway", "secondary"}, {"oneway", "yes"}, {"osm_id", "1"}}},
+      {"EC", {{"highway", "secondary"}, {"oneway", "yes"}, {"osm_id", "2"}}},
+      {"CB", {{"highway", "primary"}, {"oneway", "yes"}, {"osm_id", "3"}}},
+      {"BA", {{"highway", "primary"}, {"oneway", "yes"}, {"osm_id", "3"}}},
+      {"DC", {{"highway", "primary"}, {"oneway", "yes"}, {"osm_id", "3"}}},
+  };
+
+  const gurka::relations relations = {
+      {{
+           {gurka::way_member, "FE", "from"},
+           {gurka::way_member, "EC", "via"},
+           {gurka::way_member, "CB", "to"},
+       },
+       {
+           {"type", "restriction"},
+           {"restriction", "no_left_turn"},
+       }},
+  };
```

Comment:
So we have `FECB` restriction and `ABCD` have the same `way_id`.
Let's say we're starting with `B` node and trying to match `[3, 2, 1]` way_ids.
- Step 1: we go towards node `A` as `way_id(BA) == 3`. But this is wrong direction so we fallback to node `B`
- Step 2: we go towards node `D` as `way_id(BCD) == 3`. BUt this is wrong direction as well.
- Step 3: The `GetGraphIds` function return `no match found` but it must return `BCEF`